### DISCUSSION TITLE
[Question] Is Anyone Currently Working on Ocean Backport Features?

### DISCUSSION THREAD

**Discussion by Rozmir-Rohi** [Roadhog360/Et-Futurum-Requiem#444] [Ideas]
Hi,

Is anyone currently working on ocean backport features for Et Futurum Requiem such as depth strider enchantment, coral, kelp, sea grass, and sea pickles? Or have any of these features already been added?

I was planning to make a mod a few months from now that will add these features (which could also possibly be merged into Et Futurum Requiem after it is completed), and I wanted to ask this before making future plans.

Thanks,
Rozmir

**Comment by Roadhog360** [comment]
Currently only depth strider is planned. I  need to figure out fake waterlogged rendering to do the majority of ocean content.

**Reply by Rozmir-Rohi** [reply]
ok, I will skip out on the depth strider enchantment.

Instead, I will try to backport coral, kelp, sea grass and sea pickles in my mod.

**Reply by Rozmir-Rohi** [reply]
> ok, I will skip out on the depth strider enchantment.
> 
> Instead, I will try to backport coral, kelp, sea grass and sea pickles in my mod.

nevermind, I quit this project.

**Reply by Roadhog360** [reply]
Hard ain't it?

**Reply by DarianLStephens** [reply]
For waterlogged stuff, I would again recommend [Fluidlogged API](https://modrinth.com/mod/fluidlogged-api) for 1.12.
It adds logging any block with any fluid (both configurable), and may be a decent starting point for a 1.7 version.
We've got [ChunkAPI](https://modrinth.com/mod/chunkapi) which may be a substitute for chunk capabilities, and is being used for [Right Proper Lighting Engine](https://modrinth.com/mod/rple).

**Reply by Roadhog360** [reply]
1.12 is a better starting point because it has blockstates, 1.7 doesn't

**Reply by WyattTheSkid** [reply]
Why? No time? Just don't want to finish it? If you had anything I'd be happy to review what you have and try to finish it. A good starting point would be to look at biomes o' plenty's code for 1.7 and see how their kelp and coral and all that stuff works and adapt your own version. Just be careful not to copy pasta anything from that project specifically as their license pretty much doesn't want anyone to do anything with any of its code.
> > ok, I will skip out on the depth strider enchantment.
> 
> > 
> 
> > Instead, I will try to backport coral, kelp, sea grass and sea pickles in my mod.
> 
> 
> 
> nevermind, I quit this project.

**Reply by Roadhog360** [reply]
I'd wager there wasn't much progress considering how difficult it is in the first place. The rendering gimmicks and other hacks needed is a lot of work. Trying to get a block to behave as a water source when it isn't actually a liquid has proved to be challenging for me too.

**Reply by DarianLStephens** [reply]
> 1.12 is a better starting point because it has blockstates, 1.7 doesn't

As far as I'm aware, the mod doesn't use blockstates at all. It's an entirely separate thing.

**Reply by Rozmir-Rohi** [reply]
> > ok, I will skip out on the depth strider enchantment.
> > Instead, I will try to backport coral, kelp, sea grass and sea pickles in my mod.
> 
> nevermind, I quit this project.

I never intended to backport waterlogging as it's beyond my skill level, I only meant this project to be for ocean flora.

The reason I left this project is because I'm no longer interested in it. I don't have anything to show either as I never got around to starting it.
I'm aiming to finish work on my current projects rather than start new ones now as I want to leave modding to move on to other things by the end of next February.


For anyone else who wants to take on this project, backporting kelp may be the easiest to do out of all the ocean flora as the source code for kelp and kelp generation in the mariculture mod could be used as a guide, this is also open source (MIT license). After this, you would just need to add the remaining kelp items and blocks, add the sounds, register dried kelp item as a food item, and register the dried kelp block as a fuel item.
Mariculture's coral generation could also be used as a guide for backporting coral generation.
